media/filters/chunk_demuxer.cc

```diff
@@ -767,6 +787,23 @@ ChunkDemuxer::Status ChunkDemuxer::AddId(const std::string& id,
                        ExpectedCodecs(content_type, codecs));
 }
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+ChunkDemuxer::Status ChunkDemuxer::AddId(const std::string& id,
+                                         const std::string& mime_type) {
+  String mime(mime_type);
+  blink::ContentType content_type(mime);
+  String type = content_type.GetType();
+  String codecs = content_type.Parameter("codecs");
+  if (type.empty() || codecs.empty()) {
+    return kNotSupported;
+  }
+
+  DCHECK(id_to_mime_map_.find(id) == id_to_mime_map_.end());
```

Comment:
nit: `DCHECK(base::Contains(id_to_mime_map_, id));` because is more idiomatic/easier to read.